specs/ssh_configuration.adoc

```diff
@@ -0,0 +1,149 @@
+= SSH configuration for provisioning cluster
+
+SSH public-key configuration is used to authenticate cluster nodes with
+provisioning node.
+
+== Problem description
+
+The cluster provisioning process needs some communication between the provisioning
+node and cluster nodes. Before that the provisioning node needs some authentication
+to enable the secure communication with cluster nodes. SSH public-key configuration
+is the process of authenticating cluster nodes with provisioning node.
+
+== Use Cases
+
+Needs SSH configuration utility in common to support SSH configuration for
+provisioning cluster.
+
+== Proposed change
+
+SSH public-key configuration have four major process:
```

Comment:
This is not supposed to be an all-in-one method. Each of the following steps is supposed to be a separate utility method that *always* works locally on a specific node. Each of these methods must execute ansible in local mode. The entire workflow will need to be tied together with flows, which are not part of this specification.